Dockerfile

```diff
@@ -0,0 +1,38 @@
+FROM ubuntu:21.04
+ARG DEBIAN_FRONTEND=noninteractive
+
+ENV BUILD_DEPENDENCIES="libicu-dev \
+    libfreetype6-dev \
+    libharfbuzz-dev \
+    libxml2-dev \
+    libjpeg-dev \
+    libtiff-dev \
+    libwebp-dev \
+    libcairo2-dev \
+    libproj-dev \
+    libgdal-dev \
+    libboost-filesystem-dev \
+    libboost-program-options-dev \
+    libboost-regex-dev \
+    "
+
+WORKDIR /app
+COPY . .
+
+RUN apt-get update && apt-get install -y gpg wget \
+    && wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null | gpg --dearmor - | tee /usr/share/keyrings/kitware-archive-keyring.gpg >/dev/null \
+    && echo 'deb [signed-by=/usr/share/keyrings/kitware-archive-keyring.gpg] https://apt.kitware.com/ubuntu/ focal main' | tee /etc/apt/sources.list.d/kitware.list >/dev/null \
+    && apt-get update \
+    && apt-get install -y \
+        cmake \
+        ninja-build \
+        build-essential \
+        $BUILD_DEPENDENCIES \
+    && cmake --preset linux-gcc-release -DBUILD_DEMO_VIEWER=OFF -DBUILD_TESTING=OFF -DBUILD_DEMO_CPP=OFF -DBUILD_BENCHMARK=OFF \
```

Comment:
It might be a good idea to split this command into 2 parts: 
- installing the system dependencies
- building the project

This makes it possible to cache the Docker layer with the system dependencies. When the system dependencies do not change, the Docker build will start with the existing cached image.

The `COPY . .` should go after the system depedencies, because otherwise any change in any file in this project will invalidate the Docker layer cache and rebuild everything.